transform/all_collection/spark/transformations/commons.py

```diff
@@ -82,33 +96,65 @@ def harvest_sdg_and_fos(
         harvested_properties[prop] = harvested_prop_column
 
 
-def simplify_bestaccessright(df: DataFrame) -> DataFrame:
-    """Simplify best_access_right - get only label and convert structure to a string"""
-    return df.withColumn("bestaccessright", col("bestaccessright")["label"])
+def harvest_best_access_right(
+    df: DataFrame, harvested_properties: Dict, col_name: str
+) -> DataFrame:
+    """Harvest best_access_right and map standardize its value"""
+    if col_name.lower() in {"dataset", "publication", "software"}:
+        df = df.withColumn(BEST_ACCESS_RIGHT, col(BEST_ACCESS_RIGHT)["label"])
+
+    # Values are mapped to the keys
+    mapping = {
+        _OPEN_ACCESS: (
+            "OPEN",
+            "open_access",
+            "fully_open_access",
+            "open access",
+            "free access",
+            "free access ",
+        ),
+        RESTRICTED: ("RESTRICTED",),
+        ORDER_REQUIRED: ("order_required",),
+        LOGIN_REQUIRED: ("login required",),
+        LOGIN_REQUIRED_ON: (
+            "login required on EOSC Pillar, open access on the original resource page",
+        ),
+        CLOSED: ("CLOSED",),
+        EMBARGO: ("EMBARGO",),
+        OTHER: ("other",),
+    }
+    best_access_right = df.select(BEST_ACCESS_RIGHT).collect()
+    best_access_right_column = []
+
+    for access in best_access_right:
+        if not access[BEST_ACCESS_RIGHT]:
+            best_access_right_column.append(None)
+            continue
+
+        for desired_access_t, access_t in mapping.items():
+            if access[BEST_ACCESS_RIGHT] in access_t:
+                best_access_right_column.append(desired_access_t)
+                break
+        else:
+            print(
+                f"Warning unknown access right: best_access_right={access[BEST_ACCESS_RIGHT]}, collection={col_name}"
+            )
+            best_access_right_column.append(access[BEST_ACCESS_RIGHT])
```

Comment:
I'm not sure with which `if` works the `else` since indenting isn't obvious.

Reply:
This else is for the` for loop` :sunglasses: https://www.geeksforgeeks.org/using-else-conditional-statement-with-for-loop-in-python/

Reply:
Thank you for showing me some new crazy stuff in the python syntax. :)